core/src/main/java/com/linecorp/armeria/server/logging/Sampler.java

```diff
@@ -83,6 +83,11 @@ public String toString() {
      * @param rate minimum sample rate is 0.01, or 1% of traces
      */
     static Sampler create(float rate) {
+        if (rate == 1.0) {
```

Comment:
Oops - nah I missed `CountingSampler.create` ;) I think it's weird for an implementation's factory to delegate to other implementations of the interface, and went ahead and replaced `CountingSampler.create` with `Sampler.create`. Hope it makes sense.